src/request.rs

```diff
@@ -73,31 +76,32 @@ pub trait DispatchSignedRequest {
 
 impl DispatchSignedRequest for Client {
     fn dispatch(&self, request: &SignedRequest) -> Result<HttpResponse, HttpDispatchError> {
-        let hyper_method = match request.method().as_ref() {
-            "POST" => Method::Post,
-            "PUT" => Method::Put,
-            "DELETE" => Method::Delete,
-            "GET" => Method::Get,
-            "HEAD" => Method::Head,
-            v => return Err(HttpDispatchError { message: format!("Unsupported HTTP verb {}", v) })
+        // TODO: be more graceful for using the builder:
+        let hyper_method = match Method::from_str(request.method().as_ref()) {
```

Comment:
Might want to stop calling this `hyper_method` since Hyper isn't involved any more.